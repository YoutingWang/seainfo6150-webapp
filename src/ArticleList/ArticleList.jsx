import React, { useState } from "react";
import PropTypes from "prop-types";
import ArticleListItem from "../ArticleListItem/ArticleListItem.jsx";

const ArticleList = (props) => {
  return (
    <div>
       <ul>
          {props.articles.map((article) => (
            <li key={article.slug}>
             <ArticleListItem article = {article}/>
            </li>
          ))}
        </ul>
    </div>
  );
};

ArticleList.propTypes = {
  articles: PropTypes.array.isRequired
};

export default ArticleList