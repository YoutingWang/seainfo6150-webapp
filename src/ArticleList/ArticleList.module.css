.articleList {
    background-color: #9b9999;
    box-shadow: red 0px 0px 15px;
    border: 2px #828485 dotted;
    border-radius: 12px;
    margin: 25px;
    padding: 15px;
    list-style: none;
    align-items: stretch;
    column-gap: 15px;
    row-gap: 5px;
  }
  .articleGrid {
    display: grid;
    grid-auto-flow: row;
  } 

  @supports (display: grid) {
    .articleGrid {
      display: grid;
      grid-auto-flow: row;
    }
    .articleList {
      background-color: #9b9999;
      border: 2px #828485 dotted;
      list-style: none;
      display: grid;
      grid-auto-flow: row;
      grid-template-rows: auto;
      align-items: stretch;
      box-shadow: red 0px 0px 15px;
      column-gap: 15px;
      row-gap: 2px;
      border-radius: 12px;
      margin: 15px;
      padding: 10px;
      grid-template-columns: 1fr; 
      font-size: 105%;
    }
    @media screen and (min-width: 550px) {
      .articleList {
        grid-template-columns: 1fr 1fr; 
        font-size: 100%;
      }
    }
    @media screen and (min-width: 900px) {
      .articleList {
        grid-template-columns: 1fr 1fr 1fr; 
        font-size: 90%;
      }
    }
    @media screen and (min-width: 1300px) {
      .articleList {
        grid-template-columns: 1fr 1fr 1fr 1fr; 
        font-size: 80%;
      }
    }
  } 