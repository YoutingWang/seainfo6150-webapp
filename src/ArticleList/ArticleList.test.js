import React from "react";
import ArticleList from "./ArticleList";

describe("ArticleList tests", () => {
            it("renders correctly", () => {
                    const articles = [{
                            slug: "181122102925-statue-of-liberty-crown",
                            shortText: "The Statue of Liberty's original torch is getting a new home",
                            title: "The Statue of Liberty's torch heads to new museum",
                            displayDate: "November 22nd 2018, 7:12 am"
                        },
                        {
                            slug: "170321135754-airport-security-line",
                            shortText: "Getting excited to eat your mother's roast turkey, your uncle's stuffing and your sister's famous green bean casserole?",
                            title: "You can take a pie on a plane",
                            displayDate: "November 21st 2018, 2:57 pm"
                        },
                        {
                            slug: "181119170125-ploughgate-creamery-butter-ball",
                            shortText: "Amid the farm houses and barns dotting the Mad River Valley area of rural Vermont, five miles north of Sugarbush ski resort, Marisa Mauro is making butter.",
                            title: "Butter is back, just in time for the holidays",
                            displayDate: "November 21st 2018, 5:17 am"
                        },
                        {
                            slug: "181121164246-boeing-777-9x-assembly-1",
                            shortText: "Travelers around the globe are one step closer to seeing a plane with its wings folded up being taxied to their gate.",
                            title: "Boeing's 777X jetliner comes together ",
                            displayDate: "November 22nd 2018, 6:58 am"
                        }
                    ];
                    const {
                        container
                    } = render( < ArticleList articles = {
                            articles
                        }
                        />);
                        expect(container).toMatchSnapshot();
                    });
            });