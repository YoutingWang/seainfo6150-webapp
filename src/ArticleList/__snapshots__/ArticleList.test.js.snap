// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList tests renders correctly 1`] = `
<div>
  <div>
    <ul
      class="articleList"
    >
      <li
        class="articleGrid"
      >
        <article
          class="articleBody"
        >
          <h2
            class="articleTitle"
          >
            The Statue of Liberty's torch heads to new museum
          </h2>
          
          <button
            class="toggleButton"
          >
            Show more
          </button>
        </article>
      </li>
      <li
        class="articleGrid"
      >
        <article
          class="articleBody"
        >
          <h2
            class="articleTitle"
          >
            You can take a pie on a plane
          </h2>
          
          <button
            class="toggleButton"
          >
            Show more
          </button>
        </article>
      </li>
      <li
        class="articleGrid"
      >
        <article
          class="articleBody"
        >
          <h2
            class="articleTitle"
          >
            Butter is back, just in time for the holidays
          </h2>
          
          <button
            class="toggleButton"
          >
            Show more
          </button>
        </article>
      </li>
      <li
        class="articleGrid"
      >
        <article
          class="articleBody"
        >
          <h2
            class="articleTitle"
          >
            Boeing's 777X jetliner comes together 
          </h2>
          
          <button
            class="toggleButton"
          >
            Show more
          </button>
        </article>
      </li>
    </ul>
  </div>
</div>
`;
